### PR TITLE
update alpaka to latest dev

### PR DESCRIPTION
**Pull request by psychocoderHPC**
Update alpaka to pre 1.2.0 version https://github.com/alpaka-group/alpaka/tree/ab3092357bb0b917b4cc396ce49e47f7ac1924e1

```
GIT_AUTHOR_NAME="Third Party" GIT_AUTHOR_EMAIL="picongpu@hzdr.de" git subtree pull --prefix thirdParty/alpaka git@github.com:alpaka-group/alpaka.git develop --squash
```

- this pulls many bug-fixes e.g. re-enabling the usage of boost atomic-ref if available (speed-up CPU code)

- [x] update the alpaka branch against after [2348](https://github.com/alpaka-group/alpaka/pull/2348) is merged

### CODE REVIEW

**Comment by psychocoderHPC** [comment]
@ComputationalRadiationPhysics/picongpu-maintainers after the CI passed this PR can be merged.

**Review by chillenzer — APPROVED** [review]

**Pull request merged**
